src/main/java/com/couchbase/lite/Attachment.java

```diff
@@ -124,15 +125,36 @@ public String getContentType() {
     public InputStream getContent() throws CouchbaseLiteException {
         if (body != null) {
             return body;
-        }
-        else {
+        } else {
             Database db = revision.getDatabase();
-            Attachment attachment = db.getAttachmentForSequence(revision.getSequence(), this.name);
+            long sequence = getAttachmentSequence();
+            if (sequence == 0) {
+                throw new CouchbaseLiteException(Status.INTERNAL_SERVER_ERROR);
+            }
+            Attachment attachment = db.getAttachmentForSequence(sequence, this.name);
             body = attachment.getContent();
+            if (attachment.getGZipped()) {
```

Comment:
As you commented in the code, Router could expects gzipped file. Need to update this logic.
https://github.com/couchbase/couchbase-lite-java-core/blob/12c29af1b4f05aa260bfa36c6a02913d92acb47e/src/main/java/com/couchbase/lite/router/Router.java#L1646-L1681